website/blog/2018-08-27-7.0.0.md

```diff
@@ -0,0 +1,462 @@
+
+Here's a very simple example of a custom macro that swaps `line` and `column` imports with the line or column where it appears in the code:
+
+```javascript
+// line-column.macro
+module.exports = createMacro(lineColumnMacro)
+
+function lineColumnMacro({ references, babel }) {
+  references.line.forEach(referencePath => {
+    const num = referencePath.node.loc.start.line
+    referencePath.replaceWith(babel.types.numberLiteral(num))
+  })
+  references.column.forEach(referencePath => {
+    const num = referencePath.node.loc.start.column
+    referencePath.replaceWith(babel.types.numberLiteral(num))
+  })
+}
+```
+
+```javascript
+import {line, column} from './line-column.macro'
+
+console.log(`we're at ${line}:${column}`)
+console.log(`and now we're at ${line}:${column}`)
+```
+
+This will be transpiled into:
+
+```javascript
+console.log(`we're at ${3}:${32}`)
+console.log(`and now we're at ${4}:${40}`)
+```
+
+We don't need to change any config to add that custom transform! Writing and using code transforms is a fair amount easier this way. You can't do everything that a full Babel plugin can do, but we're only just getting started with this and look forward to what the community will do with this power.
+
+> Play around with the above macro example [on astexplorer.net](https://astexplorer.net/#/gist/e586bcbbf2ce35835115a7d808528c90/b64f5f025d98481ebfb93a582334c8562f7337f0). See if you can make it support an import called `lineColumn` that is replaced with the sum of line and column. Just for fun!
+
+Learn more about `babel-plugin-macros` in our post ["Zero-config code transformation with babel-plugin-macros"](https://babeljs.io/blog/2017/09/11/zero-config-with-babel-macros).
+
+### `class C extends HTMLElement {}`
+
+> One of our oldest issues gets it's own heading ([details](https://github.com/babel/babel/pull/7020))
+
+Babel has always had the caveat where it could not support extending native built-ins (`Array`, `Error`, etc) and now it can! We've gotten a lot of issues about this; 🎉 you should celebrate like [Andrea](https://twitter.com/WebReflection/status/940724280035303430)!
+
+This change was made to the class plugin so it should be automatically enabled if you are using `preset-env`.
+
+### Website Changes 🌏
+
+We've moved our site from [Jekyll](https://jekyllrb.com/) to [Docusaurus](https://docusaurus.io/)!
+
+We're still in the process of setting up [translations](https://github.com/babel/website/issues/1619) with [Crowdin](https://crowdin.com/project/babeljs), and with Babel 7 out, we will be in a better place to start that process.
+
+#### REPL
+
+![](https://i.imgur.com/j0qOllj.png)
+
+We've rewritten the [REPL](https://babeljs.io/repl) as a [React Component](https://github.com/babel/website/tree/master/js/repl), and have been working with Ives on integrating better with [CodeSandbox](https://codesandbox.io/).
```

Comment:
Maybe it would be good to also add something like "This allows you to install any plugin or preset from npm in the REPL". That gives the main advantage of why it's been added.

Reply:
good point! and I guess we also get the updates you put out!

Reply:
Yes 😄